### PR TITLE
Load Neighborhood geojson on map

### LINKED ISSUE #247: Implement map on homepage
Static map showing all cities, geojson probably fine

Depends on #246

### PR DESCRIPTION
**Pull request by CloudNiner**
## Overview

Part 2 after #316. Adds the neighborhoods as geojson points to the map.

- Not styled
- Link to neighborhood detail view not hooked up. Can be done as part of #254 

@designmatty we'll need to think about how we want to style the markers/popups in this view

### Demo

screen shot 2017-04-19 at 14 40 00

### Notes

I intentionally left the map bounds set to CONUS because the final application will have points across the US.

Refactors the neighborhood map logic into a self-contained directive

Closes #247

### CODE REVIEW

**Comment by CloudNiner** [comment]
Whoops. Realized I didn't implement the count in the map overlay callout. Working on that now.

**Comment by CloudNiner** [comment]
Updated!

**Comment by flibbertigibbet** [comment]
I'm seeing locations as points and not boundaries. Any idea why that is?

File: src/angularjs/src/app/home/neighborhood-map.directive.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            });
+
+            function onEachFeature(feature, layer) {
+                // TODO: Style marker and popup
+                // TODO: Add link to neighborhood detail in popup
+                layer.on({
+                    click: function () {
+                        var popup = L.popup()
+                            .setLatLng([
+                                feature.geometry.coordinates[1],
```
Comment (flibbertigibbet) [inline src/angularjs/src/app/home/neighborhood-map.directive.js]: Any chance of having a missing or broken geometry here we'd want to guard against?

Comment (CloudNiner) [inline src/angularjs/src/app/home/neighborhood-map.directive.js]: Hmm, would hurt to guard. Can add that.

**Comment by CloudNiner** [comment]
Yeah, neighborhoods are points because we only return centroids from the endpoint.

**Pull request merged**